### ISSUE TITLE
nhidden of the last fc layer

### ISSUE THREAD

**Issue by lsy1993311** [dmlc/cxxnet#150]
when I change MNIST.conf's netconfig to be:
netconfig=start
layer[0->1] = fullc:fc1
  nhidden = 100
  init_sigma = 0.01
layer[1->2] = batch_norm:bn1
layer[2->sg1] = sigmoid:se1
layer[sg1->fc2] = fullc:fc2
  nhidden = 100
  init_sigma = 0.01
layer[fc2->bn2] = batch_norm:bn2
layer[bn2->sg2] = sigmoid:se2
layer[sg2->fc3] = fullc:fc3
  nhidden = 100
  init_sigma = 0.01
layer[+0] = softmax
netconfig=end

note that I use nhidden=100 in the last fc layer, but I still got good result, I think it's a bug.

**Comment by antinucleon** [comment]
It is not a bug. If you check how the gradient is out and how the gradient update the net, you will know why it works.

**Comment by lsy1993311** [comment]
Ok, I'll check。
